glib/src/gstring.rs

```diff
@@ -392,44 +730,187 @@ impl GString {
 
     // rustdoc-stripper-ignore-next
     /// Extracts the [`GStr`] containing the entire string.
+    #[inline]
     pub fn as_gstr(&self) -> &GStr {
         let bytes = match self.0 {
-            Inner::Native(ref cstr) => cstr.as_ref().unwrap().to_bytes_with_nul(),
+            Inner::Native(None) => return <&GStr>::default(),
+            Inner::Native(Some(ref s)) => s.as_bytes(),
             Inner::Foreign { len, .. } if len == 0 => &[0],
```

Comment:
Could combine this and the first branch I guess? But maybe we can forbid `Native(None)` for simplification and only have one possible representation of an empty string